enhancements/cluster-logging/elasticsearch-cert-management.md

```diff
@@ -0,0 +1,158 @@
+---
+title: elasticsearch-cert-management
+authors:
+  - "@ewolinetz"
+reviewers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+  - "@jpkrohling"
+approvers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+  - "@jpkrohling"
+creation-date: 2021-01-05
+last-updated: 2021-01-05
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# elasticsearch-cert-management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Migration plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently to use the Elasticsearch Operator, one would need to generate secrets in a specific manner which creates a high amount of complexity for use. This proposal seeks to outline a mechanism where the Elasticsearch Operator creates and maintains these certificates and the elasticsearch/kibana secret instead of other operators (e.g. Cluster Logging and Jaeger), and allows an annotation mechanism on secrets for injecting required keys and certificates for mTLS with Elasticsearch.
+
+## Motivation
+
+Currently our managed Elasticsearch cluster requires anything using our operator to generate, manage, and provide certs and secrets for the operator and its cluster to use. This leads to many potential sources to create certificates and can cause a gap between what the managed Elasticsearch requires and what something may provide. This would instead make the Elasticsearch Operator a single source of truth for how certificates would be generated.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the responsibility for the Elasticsearch Operator regarding certificates
+
+* Expose the desired annotation that the operator would watch for on secrets where an elasticsearch CR exists
```

Comment:
on secrets in the namespace where an elasticsearch CR exists?

Reply:
this is a little clunky, i'll rephrase. essentially the operator will watch secrets in namespaces where someone has created an elasticsearch CR. If it has the annotation then we'll perform the cert generation and then inject it back into that secret.

Reply:
That's similar to how the service-ca works, right?

Reply:
@jpkrohling correct